### PR TITLE
fix(files): Check if file exists before renaming in cache

### PR DESCRIPTION
**Pull request by mattrobenolt**

### CODE REVIEW

File: src/sentry/models/file.py [hunk 1]
```diff
@@ -355,7 +355,18 @@ def save_to(self, path):
...
+
+            # pre-emptively check if the file already exists.
+            # this can happen as a race condition if two processes/threads
+            # are trying to cache the same file and both try to write
+            # at the same time, overwriting each other. Normally this is fine,
+            # but can cause an issue if another process has opened the file
+            # for reading, then the file that was being read gets clobbered.
+            # I don't know if this affects normal filesystems, but it
+            # definitely has an issue if the filesystem is NFS.
+            if not os.path.exists(path):
```
Comment (JTCunning) [inline src/sentry/models/file.py:367]: Should we have an `else` removing the file so it doesn't just stick around in tmp?

Comment (mattrobenolt) [reply src/sentry/models/file.py:367]: It's already being done in `finally` path.

Comment (mattrobenolt) [reply src/sentry/models/file.py:367]: Oh, I see. I guess it would be orphaned. Lemme fix up.

Comment (mattrobenolt) [reply src/sentry/models/file.py:367]: Fixed so it does rely on the finally path now.

**Review by JTCunning — APPROVED** [review]
🍪

**Pull request merged**